lib/misc.coffee

```diff
@@ -1,181 +1,51 @@
-_ = require 'underscore'
-
-# Function that translates an index from start (absolute position) into a
-# relative position in word index and offset
-#
-# @param [Integer] position the position
-# @param [Y.RichText] RichText a richText instance
-# @return [Object] options returning object
-# @option options [Integer] word the index of the word
-# @option options [Integer] position the offset in this word
-relativeFromAbsolute = (position, richText)->
-  index = 0
-  while position > 0
-    if index >= richText.getWords(0).length
-      index-- #position = 0
-      break
-    if richText.getWord(index).word.length > position
-      break
-    else
-      position -= richText.getWord(index).word.length
-      index++
-
-  return {word: index, pos: position}
-
-# Function that translates an index from start (absolute position) into a
-# relative position in word index and offset
-#
-# @param [Option] relative the position
-# @param [Y.RichText] richText an richText instance
-# @return [Integer] the absolute index
-absoluteFromRelative = (index, offset, richText) ->
-  absolute = offset
-  if index > 0
-    for i in [0..(index-1)]
-      absolute += richText.getWord(i).word.length
-
-  absolute
-
-# Initialize an Yjs list and add common array operations
-# over it
-customList = (Operation) ->
-  ret = new Operation.ListManager(@).execute()
-
-  # Another name for delete is slice
-  # TODO: remove all occurrences of slice!
-  ret.slice = (start, end) ->
-    @delete start, end
-  ret.splice = (start, end, insert...) ->
-    @delete start, end
-    @insert start, insert
-  ret.indexOf = (element) ->
-    @val().indexOf(element)
-
-  ret.observe = (e) ->
-    console.log e
-
-  return ret
-
 
 # a basic class with generic getter / setter function
 class BaseClass
```

Comment:
I really like the idea of the baseclass. I'm thinking about of putting it in a helper directory in Yjs, so that other developers can use it too. I changed it to prevent possible pitfalls